wp/wp-content/themes/lahma_maker/js/scripts.js

```diff
@@ -291,16 +291,58 @@ $( document ).ready(function() {
     jQuery(document).on('click', '#sortreverse', function(e) {
         e.preventDefault()
 
-        jQuery('.arcsi-blokks').html(jQuery('.arcsi-blokks').find(jQuery('.arcsi-blokks .arcsiblokk')).get().reverse());
+        const mylist = $('.arcsi-blokks');
+        let listitems = mylist.children(".arcsiblokk");
 
         if (ascending) {
             ascending = false
             $(this).find('i').removeClass('fa-sort-numeric-asc').addClass('fa-sort-numeric-desc')
+            listitems.sort(function(b, a) {
+            let compA = $(a).find('.airtime').text().toUpperCase();
+            let compB = $(b).find('.airtime').text().toUpperCase();
+                return (compA < compB) ? -1 : (compA > compB) ? 1 : 0;
+            })
+            $(mylist).append(listitems);  
         } else {
             ascending = true
             $(this).find('i').removeClass('fa-sort-numeric-desc').addClass('fa-sort-numeric-asc')
+             listitems.sort(function(a, b) {
+            let compA = $(a).find('.airtime').text().toUpperCase();
+            let compB = $(b).find('.airtime').text().toUpperCase();
+                return (compA < compB) ? -1 : (compA > compB) ? 1 : 0;
+            })
+            $(mylist).append(listitems);           
         }
     })
 
+    let alphabetascending = true
+    jQuery(document).on('click', '#alphabetical', function(e) {
+        e.preventDefault()
+
+        const mylist = $('.arcsi-blokks');
+        let listitems = mylist.children(".arcsiblokk");
+
+        if (alphabetascending) {
+            alphabetascending = false
+            $(this).find('i').removeClass('fa-sort-alpha-asc').addClass('fa-sort-alpha-desc')
+            listitems.sort(function(b, a) {
+            let compA = $(a).find('h4').text().toUpperCase();
+            let compB = $(b).find('h4').text().toUpperCase();
+                return (compA < compB) ? -1 : (compA > compB) ? 1 : 0;
+            })
+            $(mylist).append(listitems);
+        } else {
+            alphabetascending = true
+            $(this).find('i').removeClass('fa-sort-alpha-desc').addClass('fa-sort-alpha-asc')
+            listitems.sort(function(a, b) {
+            let compA = $(a).find('h4').text().toUpperCase();
+            let compB = $(b).find('h4').text().toUpperCase();
+                return (compA < compB) ? -1 : (compA > compB) ? 1 : 0;
+            })
+            $(mylist).append(listitems);
```

Comment:
Later improvement could be if we introduce maybe many more sorting opts to create a named function that has sorting option (date, alphabet, etc.) as its parameter, then we only need to pass it once and the whole function just sorts once and appends 1 final listitems at the very end of this logic. We only render 1 list in the end anyway.

This is just general thought on the next steps, from what I can tell the code looks good, works as intended! Good job thanks!

Reply:
I agree, it's a general antipattern to use inplace functions (as long as performance is not an issue). At the moment we copy the same function four times.

Reply:
Ok, could be refucktored, but for now, for this two cases, it is OK like this...

Reply:
It's not two cases, it's four ;)